src/main/java/io/fabric8/jenkins/openshiftsync/BuildConfigToJobMap.java

```diff
@@ -36,6 +45,7 @@ static synchronized void initializeBuildConfigToJobMap() {
                 }
             }
         }
+        reconcileRunsAndBuilds();
```

Comment:
So only calling here means we are only going to prune job runs on startup.

Not a bad thing necessarily, but when I head this discussed during planning, I had taken the implication that we were going to poll in the background, and try to keep job runs in line on a periodic basis, especially since Jenkins might be running for a long time.

Was there some discussion around this that I missed?

Reply:
This basically makes sure that runs/builds get reconciled if jenkins goes down and comes back up.  There is another spot that deletes the run when a build is removed.  If you want it to run intermittently, I can find a place to put it, but I think we should still keep the code that deletes a run when a build is removed.

Reply:
To be consistent with watch/relist model, where the periodic relist compensates for missed watch events, we should probably cross reference the list of builds we get from the relist (onInitialBuilds, etc. that is on the 5 minute poll interval) with the list of runs for a given job ... if a job run does not have a corresponding build, delete it.

Sorry, my earlier comment was not clear on leveraging the existing build relist/poll.

Reply:
+1, reconciliation needs to be done periodically(on the relist interval) as there's no guarantee we didn't miss a delete event, even if jenkins wasn't down/restarted.